catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/SetVariableTest.java

```diff
@@ -0,0 +1,158 @@
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.hasSibling;
+import static android.support.test.espresso.matcher.ViewMatchers.isClickable;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withChild;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withParent;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.deleteItemWithTextFromFormulaEditorDataList;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.core.StringStartsWith.startsWith;
+
+@RunWith(AndroidJUnit4.class)
+public class SetVariableTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	public void createProject() {
+		SetVariableBrick setVariableTestBrick1 = new SetVariableBrick(10);
+		SetVariableBrick setVariableTestBrick2 = new SetVariableBrick(0);
+		Script script = BrickTestUtils.createProjectAndGetStartScript("setVariableTest");
+		script.addBrick(setVariableTestBrick1);
+		script.addBrick(setVariableTestBrick2);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testCreateNewUserVariableAndDeletion() {
+		String userVariableName = "testVariable1";
+		String secondUserVariableName = "testVariable2";
+
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_set_variable);
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(userVariableName);
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(secondUserVariableName);
+		onBrickAtPosition(1).onFormulaTextField(R.id.brick_set_variable_edit_text)
+				.perform(click());
+
+		deleteItemWithTextFromFormulaEditorDataList(userVariableName);
+		pressBack();
+		onView(withText(userVariableName))
+				.check(doesNotExist());
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).onChildView(withText(userVariableName))
+				.check(doesNotExist());
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).onChildView(withText(secondUserVariableName))
+				.check(matches(isDisplayed()));
+	}
+
+
+	//TODO: Move this to a Formula Editor Test, which?
```

Comment:
https://github.com/Catrobat/Catroid/pull/2656/files also includes this already.

if ever in doubt, feel free to add a test class in a package you deem more appropriate, but do not have TODO comments in PRs im not going to merge them ever...